backend/client_api.c

```diff
@@ -1129,6 +1129,11 @@ void error(char *msg) {
 
 int send_packet(void * buf, unsigned len, int sockfd)
 {
+	if(sockfd == 0)
```

Comment:
Is this even necessary though?

I initially added an `assert(sockfd != 0);` here so we'd just exit in these situations rather than continue... but after changing sockfd to -1 everywhere I removed the assert since -1, unlike 0, is actually an invalid fd and the write() a few lines down will catch it with its error handling.

Reply:
Right, I was thinking to specifically catch sockfd value setting errors in the spirit of this PR, to specifically see if we update the fd properly everywhere, since the write() below could error for other reasons than the sock being invalid. I agree it's optional.

Reply:
Okay, then I vote for an assert which I think is more explicit and since it's clearly indicative of coding error we should just exit and not continue with a bad state...

Reply:
OK!